### PR TITLE
style: Use an enumerated array for per-pseudo maps.

### PR DESCRIPTION
**Pull request by emilio**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17999)

### CODE REVIEW

**Comment by emilio** [comment]
This is on top of #17992.

r? @heycam 

(Needs the servo bits I'll finish off tomorrow, but mod that it should be ready to go).

File: components/style/stylist.rs [hunk 1]
```diff
@@ -240,7 +240,7 @@ impl Stylist {
 
             cascade_data: CascadeData::new(),
             animations: Default::default(),
-            precomputed_pseudo_element_decls: Default::default(),
+            precomputed_pseudo_element_decls: PerPseudoElementMap::default(),
```
Comment (heycam) [inline components/style/stylist.rs]: Just `Default::default()`?

File: components/style/stylist.rs [hunk 2]
```diff
@@ -1666,14 +1666,14 @@ struct PerOriginCascadeData {
...
-    pseudos_map: FnvHashMap<PseudoElement, SelectorMap<Rule>>,
+    pseudos_map: PerPseudoElementMap<SelectorMap<Rule>>,
 }
 
 impl PerOriginCascadeData {
     fn new() -> Self {
         Self {
             element_map: SelectorMap::new(),
-            pseudos_map: Default::default(),
+            pseudos_map: PerPseudoElementMap::default(),
```
Comment (heycam) [inline components/style/stylist.rs]: Here too?  (Unless there's a reason it needed to change...)

**Review by heycam — APPROVED** [review]
Looks good.

**Pull request merged**